Branch-SDK/src/io/branch/referral/ServerRequestRActionCompleted.java

```diff
@@ -0,0 +1,115 @@
+package io.branch.referral;
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+import android.util.Log;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import io.branch.referral.util.CommerceEvent;
+
+/**
+ * <p>
+ * The server request for Action completed event. Handles request creation and execution.
+ * </p>
+ */
+class ServerRequestRActionCompleted extends ServerRequest {
+
+    private final BranchViewHandler.IBranchViewEvents callback_;
+
+    /**
+     * <p>Creates an ActionCompleteRequest instance. This request take care of reporting specific user
+     * actions to Branch API, with additional app-defined meta data to go along with that action.</p>
+     *
+     * @param context  Current {@link Application} context
+     * @param action   A {@link String} value to be passed as an action that the user has carried
+     *                 out. For example "registered" or "logged in".
+     * @param metadata A {@link JSONObject} containing app-defined meta-data to be attached to a
+     *                 user action that has just been completed.
+     */
+    public ServerRequestRActionCompleted(Context context, CommerceEvent commerceEvent, JSONObject metadata, BranchViewHandler.IBranchViewEvents callback) {
+        super(context, Defines.RequestPath.CompletedAction.getPath());
+        callback_ = callback;
+        JSONObject post = new JSONObject();
+
+        try {
+            post.put(Defines.Jsonkey.IdentityID.getKey(), prefHelper_.getIdentityID());
+            post.put(Defines.Jsonkey.DeviceFingerprintID.getKey(), prefHelper_.getDeviceFingerPrintID());
+            post.put(Defines.Jsonkey.SessionID.getKey(), prefHelper_.getSessionID());
+            if (!prefHelper_.getLinkClickID().equals(PrefHelper.NO_STRING_VALUE)) {
```

Comment:
We want to pass the link click ID in this post? Is this for analytics purposes? cc @derrickstaten -- just confirming this is required